scripts/lib/setup/installers.sh

```diff
@@ -1,406 +1,64 @@
-        -o $destination
-
-      chmod 755 $destination
-
-      cd $HOME && lein downgrade $required_version
-EOF
-  else
-    already_installed "lein"
-  fi
-}
-
-function install_clojure_cli() {
-  if is_macos; then
-    local required_version=$(toolversion clojure_cli)
-    brew_install clojure $required_version 90ea0cb4b194282b5906108dcec522c5a1ed7ce0
-  elif is_linux; then
-    install_clojure_cli_linux
-  fi
-}
-
-function install_clojure_cli_linux() {
-  local required_version=$(toolversion clojure_cli)
+function install_nix() {
+  if ! program_exists nix; then
+    touch -a "${HOME}/.bash_profile"
 
-  if ! program_exists "clojure" || ! echo "$(clj -r <<< '(System/exit 0)')" | grep -q -o ${required_version%.*}; then
-    cecho "@b@blue[[+ Installing Clojure CLI...]]"
+    local required_version=$(toolversion nix)
+    bash <(curl https://nixos.org/releases/nix/nix-${required_version}/install) --no-daemon
+    if [ $? -eq 0 ]; then
+      echo -e "${YELLOW}**********************************************************************************************************"
+      echo "The Nix package manager was successfully installed. Please enter the following commands into the terminal:"
+      echo "  - . ~/.nix-profile/etc/profile.d/nix.sh"
+      echo "  - make shell"
+      echo "If this doesn't work, you might have to sign out and back in, in order for the environment to be reloaded."
+      echo -e "**********************************************************************************************************${NC}"
 
-    local current_dir=$(pwd)
-    sudo su << EOF
-      curl --silent \
-        https://download.clojure.org/install/linux-install-${required_version}.sh \
-        -o /tmp/clojure
-
-      chmod +x /tmp/clojure
-      cd /tmp
-      ./clojure
-EOF
-    cd "$current_dir"
-  else
-    already_installed "Clojure CLI"
-  fi
-}
-
-function install_watchman() {
-  local required_version=$(toolversion watchman)
-  if is_macos; then
-    brew_install watchman $required_version 1a41406af8db6bbc0c94811cf60043a7436be3c4
-  elif is_linux; then
-    if ! program_version_exists "watchman" "$required_version"; then
-      local current_dir=$(pwd)
-      local clone_path="/tmp/watchman"
-
-      rm -rf $clone_path
-      git clone https://github.com/facebook/watchman.git $clone_path
-      cd $clone_path
-      git checkout v$required_version
-      ./autogen.sh && \
-        ./configure && \
-        make && \
-        sudo make install
-
-      cd "$current_dir"
+      if is_linux && [ -f ~/.bashrc ]; then
+        # For some reason, new terminals are not started as login shells, so .profile and .bash_profile are not sourced.
+        # To get around it, we add Nix initialization to .bashrc as well, if it exists
+        echo "if [ -e ${HOME}/.nix-profile/etc/profile.d/nix.sh ]; then . ${HOME}/.nix-profile/etc/profile.d/nix.sh; fi # added by `make setup` Status script" >> ~/.bashrc
```

Comment:
This is a bug. By doing backticks inside of double quotes in which you have `make setup` you case an infinite loop of `make setup` being called by `make setup` being called by `make setup` being called by `make setup` being called...

Reply:
Good catch! We don't need those backticks there